core/src/test/java/org/infinispan/api/flags/FlagsEnabledTest.java

```diff
@@ -127,9 +128,19 @@ public void testReplicateSkipCacheLoad(Method m) {
       final String v = v(m, 1);
       final String k = k(m, 1);
       cache1.withFlags(Flag.SKIP_CACHE_LOAD).put(k, v);
-      // The write-skew check tries to load it from persistence.
-      assertLoadsAndReset(cache1, isTxCache() ? 1 : 0, cache2, 0);
 
+      Address primaryLocation = cache1.getDistributionManager().getPrimaryLocation(k);
```

Comment:
`SyncConsistentHashFactory` should become the default CH factory soon, and it will produce random mappings. 

I don't like tests that implicitly rely on a specific location for their keys either - especially if it means the test is meant to cover multiple cases, but only covers one, like with {{AtomicMapAPITest}}. 

But when the test explicitly requires knowing the specific location of its keys, as is the case here, I find it easier to keep the mapping constant and generate random keys, than to keep the key constant and decide what to do based on the random mapping. Or better yet, use a `ControlledConsistentHashFactory` so there's no randomness involved.